pages/1_orders_and_revenue.py

```diff
@@ -16,51 +20,330 @@
-    with row1_col1:
-        st.markdown("**Total Revenue Over Time**")
+    # Product Revenue and New Customers charts 
+    col1, col2 = st.columns(2)
+    
+    with col1:
+        st.markdown("**Product By Revenue**")
+        product_revenue = data.groupby('product_name')['total_amount'].sum().reset_index()
+        product_revenue = product_revenue.sort_values(by='total_amount', ascending=True)
+        
+        # Create the figure manually
+        fig = go.Figure()
+        
+        for index, row in product_revenue.iterrows():
+            fig.add_trace(go.Bar(
+                y=[row['product_name']],
+                x=[row['total_amount']],
+                orientation='h',
+                name=row['product_name'],
+                text=f"${row['total_amount']:,.0f}",
+                textposition='outside',
+                marker_color=px.colors.qualitative.Set3[index % len(px.colors.qualitative.Set3)]
+            ))
+        
+        fig.update_layout(
+            xaxis_title="Total Revenue",
+            yaxis_title="Product",
+            showlegend=False,
+            xaxis=dict(tickprefix='$', tickformat='~s'),
+            height=400,
+            margin=dict(l=0, r=100, t=30, b=0),  # Increased right margin
+            uniformtext_minsize=8,
+            uniformtext_mode='hide'
+        )
+        
+        # Adjust x-axis range to accommodate labels
+        max_revenue = product_revenue['total_amount'].max()
+        fig.update_xaxes(range=[0, max_revenue * 1.2])  # Extend x-axis by 20%
+        
+        st.plotly_chart(fig, use_container_width=True)
+    
+    with col2:
+        st.markdown("**New Customers Over Time**")
+        
+        # Apply the same date filter as used in other charts
+        filtered_data = data[(data['created_at'] >= min_created_at) & (data['created_at'] <= max_created_at)]
+        
+        filtered_data['customer_created_month'] = filtered_data['customer_created_at'].dt.to_period('M').dt.to_timestamp()
+        new_customers_over_time = filtered_data.groupby('customer_created_month')['customer_id'].nunique().reset_index()
+        
+        fig = px.bar(new_customers_over_time, 
+                    x='customer_created_month', 
+                    y='customer_id',
+                    color_discrete_sequence=['#2ca02c'])  # Green color for consistency
+        
+        fig.update_yaxes(title_text='New Customers', range=[0, new_customers_over_time['customer_id'].max() * 1.1])
+        fig.update_xaxes(title_text='Month', tickformat='%b %Y')
+        fig.update_traces(text=new_customers_over_time['customer_id'], textposition='outside')
+        
+        # Ensure x-axis range matches the filter
+        fig.update_xaxes(range=[min_created_at, max_created_at])
+        
+        # Update layout for consistency
+        fig.update_layout(
+            height=400  # Adjust height to match other charts if needed
+        )
+        
+        st.plotly_chart(fig, use_container_width=True)
```

Comment:
The green color looks a bit off when compared to the other blue themed charts of the page. Would you be able to switch this to blue for consistency.

Additionally, I feel New Customers All Time is more information than necessary. Can we limit this to New Customers within the date filtered time frame?

Reply:
updated the filter!